ports/stm32/mpconfigport.h

```diff
@@ -123,6 +123,10 @@
 #define MICROPY_PY_MACHINE_SOFTSPI  (1)
 #define MICROPY_PY_MACHINE_TIMER    (1)
 #endif
+#define MICROPY_PY_MACHINE_UART     (1)
+#define MICROPY_PY_MACHINE_UART_INCLUDEFILE "ports/stm32/machine_uart.c"
+#define MICROPY_PY_MACHINE_UART_IRQ (1)
+#define MICROPY_PY_MACHINE_UART_READCHAR_WRITECHAR (1)
```

Comment:
Suggest adding a block somewhere (probably `extmod/machine_uart.h`) that does `#ifndef MICROPY_PY_MACHINE_UART_READCHAR_WRITECHAR; #define MICROPY_PY_MACHINE_UART_READCHAR_WRITECHAR (0)` for all of these, and includes any relevant comments on what they are for.

(This is another suggestion that doesn't change anything at compile time, but it gives the implementer somewhere to quickly see all of the macros they may or may not want to tweak in their port.)

Reply:
Added this to `extmod/modmachine.h`.